### PR TITLE
Add B93 (TLSR9253B) architecture copied from B92 (TLSR9528A)

### PR DESCRIPTION
**Pull request by haiwentelink**
1. change the copyright to 2024 for the new added file .
2. change the B93 tag instead of the B92
3. fix the space 
4. add the ci in the telink-b93-build .yaml

### CODE REVIEW

**Comment by andriy-bilynskyy** [comment]
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/439208c46aa98297066cba042c605770c07280d8 | https://github.com/telink-semi/hal_telink/commit/f90f33dce3820aef574b0ce1b8fada3d18bb385e (develop) | [telink-semi/hal_telink@439208c4..f90f33dc](https://github.com/telink-semi/hal_telink/compare/439208c46aa98297066cba042c605770c07280d8..f90f33dce3820aef574b0ce1b8fada3d18bb385e) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

**Comment by haiwentelink** [comment]
need to merge the hal_telink first , now i use my own hal_telink , after merged i will change it back

File: soc/riscv/riscv-privilege/common/CMakeLists.txt [hunk 1]
```diff
@@ -16,3 +16,9 @@ zephyr_sources(
 	idle.c
 )
 endif()
+
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/common/CMakeLists.txt:19]: strange fork. in latest develop i see
```
zephyr_sources(
	idle.c
)
```
https://github.com/telink-semi/zephyr/blob/develop/soc/riscv/riscv-privilege/telink_b9x/CMakeLists.txt
Just ensure that you are forked from latest develop. It was changed i thing month ago. Merging old state we introduce some degradation.

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

File: soc/riscv/riscv-privilege/common/CMakeLists.txt [hunk 2]
```diff
@@ -16,3 +16,9 @@ zephyr_sources(
...
+if(!CONFIG_SOC_RISCV_TELINK_B93)
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/common/CMakeLists.txt]: Seems we already had a bug. Idle is not used for telink soc.
idle.c is inside our soc folder

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

File: soc/riscv/riscv-privilege/common/CMakeLists.txt (hunk 2 above)
Comment (s07641069) [reply soc/riscv/riscv-privilege/common/CMakeLists.txt]: @mishadesh please fix issue as common B9X

Comment (s07641069) [reply soc/riscv/riscv-privilege/common/CMakeLists.txt]: Fix added

**Review by andriy-bilynskyy — APPROVED** [review]

**Review by s07641069 — APPROVED** [review]

**Pull request merged**